app/models/concerns/foreman_bootdisk/orchestration/compute.rb

```diff
@@ -46,6 +55,16 @@ def bootdisk_attach_iso
         compute_resource.iso_attach(File.basename(bootdisk_isofile), uuid)
       end
 
+      def bootdisk_detach_iso
+        return true unless provision_method == 'bootdisk'
+        return true if previous_changes['build'].nil?
+
+        if previous_changes['build'] == [true, false]
```

Comment:
I believe we can remove these guards in this method. A developer should make sure to only call these methods when the host is in the correct state.